src/gui/Src/Gui/ZehSymbolTable.cpp

```diff
@@ -4,24 +4,35 @@
 
 class SymbolInfoWrapper
 {
-    SYMBOLINFO info;
-
-public:
-    SymbolInfoWrapper()
-    {
-        memset(&info, 0, sizeof(SYMBOLINFO));
-    }
-
-    ~SymbolInfoWrapper()
+    void free()
     {
         if(info.freeDecorated)
             BridgeFree(info.decoratedSymbol);
         if(info.freeUndecorated)
             BridgeFree(info.undecoratedSymbol);
     }
 
-    SYMBOLINFO* operator&() { return &info; }
+    SYMBOLINFO info{};
+
+public:
+    SymbolInfoWrapper() = default;
+    ~SymbolInfoWrapper() { free(); }
+
+    SymbolInfoWrapper(const SymbolInfoWrapper&) = delete;
+    SymbolInfoWrapper& operator=(const SymbolInfoWrapper&) = delete;
+
+    SYMBOLINFO* put()
+    {
+        free();
+        memset(&info, 0, sizeof(info));
+        return &info;
+    }
+
+    SYMBOLINFO* get() { return &info; }
+    const SYMBOLINFO* get() const { return &info; }
+
     SYMBOLINFO* operator->() { return &info; }
+    const SYMBOLINFO* operator->() const { return &info; }
 };

```

Comment:
This change is motivated by a mistake I made with the existing version of the wrapper - calling `GetSymbol(&info)` twice in a row may lead to a memory leak. I also think that overloading `&` is confusing, and common classes such as STL smart pointers provide a `get` method instead, which is also what I used.